SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyAllocStack.swift

```diff
@@ -179,7 +179,8 @@ private extension AllocStackInst {
   ///   use %3
   /// ```
   func optimizeExistential(_ context: SimplifyContext) -> Bool {
-    guard type.isExistential || type.isExistentialArchetype,
+    // TODO: support non-root existential archetypes
```

Comment:
I would prefer we just fixed this with the right abstractions. You can get this from the conformance of Self to the protocol, stored in the init_existential instruction.